keylime-agent/src/main.rs

```diff
@@ -170,27 +170,32 @@ async fn main() -> Result<()> {
     // Load config
     let mut config = config::KeylimeConfig::new()?;
 
+    let enable_agent_mtls = config.agent.enable_agent_mtls;
+    let enable_insecure_payload = config.agent.enable_insecure_payload;
+    let payload_script = &config.agent.payload_script;
+
     // The agent cannot run when a payload script is defined, but mTLS is disabled and insecure
     // payloads are not explicitly enabled
-    if !&config.agent.enable_agent_mtls
-        && !&config.agent.enable_insecure_payload
-        && !&config.agent.payload_script.is_empty()
+    if !enable_agent_mtls
+        && !enable_insecure_payload
+        && !payload_script.is_empty()
```

Comment:
Do we still need this change?

Reply:
Hum, I think this can be dropped.

Reply:
@THS-on Do you have any other finding? Otherwise, I'd like to merge this.

Reply:
Let me just build it and test it. But I think it is good in the current form. Thank you for fixing it!

Reply:
It works as intended :)

Reply:
Thank you for the review!